### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by gbielanova**

### CODE REVIEW

**Comment by gbielanova** [comment]
Page can be seen together with 1st hw at https://gbielanova.github.io/beetroot_hws_js/

File: js/hw2.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+function getDivider() {
+    let num1 = Number(document.getElementById('task2_num1').value);
+    let num2 = Number(document.getElementById('task2_num2').value);
+
+    let result = 0;
+
+    (num2 > num1) ? result = num1 : result = num2;
+
+    // we can add check that result > 0 but both numbers should be divided to 1
+    // but it will not work if numbers will be float, hope for the best this time
```
Comment (roskyba) [inline js/hw2.js:30]: You can `Math.floor()` floated numbers and work with integers

File: js/hw2.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    while (!(num2 % result == 0 && num1 % result == 0)) {
```
Comment (roskyba) [inline js/hw2.js:31]: The program will fail if we write negative values. Will stuck in infinite loop

File: js/hw2.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        result -= 1;
+    }
+
+    document.getElementById("task2_result").innerHTML = result;
+}
+
+function getDividers() {
+    let num = Number(document.getElementById('task3_num').value);
```
Comment (roskyba) [inline js/hw2.js:39]: What if the user will write a negative number? The code returns -1 which is not right. We should warn the user about that edge case

Or what if 0? We cannot test our program by dividing on zero, since it will output `0, -1`

And if we enter `'sdfsd34343`' in prompt we will still calculate and the result will be `NaN`, we should care about this edge case as well

File: js/hw2.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        if (num % i == 0) {
+            result += ', ' + i;
+        }
+    } while (i > 0)
+
+    document.getElementById("task3_result").innerHTML = result;
+}
+
+function getDigits() {
+    let num = Number(document.getElementById('task4_num').value);
```
Comment (roskyba) [inline js/hw2.js:57]: What if `''` empty string or `'somethingNotANumber'`? `Number(...)` will return 0 and we will get 1 as a result. This is not right, since we did not input any number

File: js/hw2.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    let result = 1;
```
Comment (roskyba) [inline js/hw2.js:58]: Or ever negative numbers. We should care as well

File: js/hw2.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+function getStats() {
+    let pos = 0;
+    let neg = 0;
+    let zeros = 0;
+    let odd = 0;
+    let even = 0;
+
+    for (let i = 0; i < 10; i++) {
+        num = Number(prompt('Введіть число'));
```
Comment (roskyba) [inline js/hw2.js:76]: Same here - if we will Number('sdfsfsdfsdfs') we will get plus more 0's. Test your code on all edge cases

File: js/hw2.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+            case ('+'):
+                result = num1 + num2;
+                break;
+            case ('-'):
+                result = num1 - num2;
+                break;
+            case ('*'):
+                result = num1 * num2;
+                break;
+            case ('/'):
```
Comment (roskyba) [inline js/hw2.js:102]: -100 / 0 = -Infinity  :)
100 / 0 = Infinity  :)

File: js/hw2.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+                break;
+            default:
+                result = 'Ця операція не підтримується'
+        }
+
+    }
+    while (confirm(`Результат: ${result}\nХочете порахувати щось ще?`));
+}
+
+function getShiftedNumber() {
```
Comment (roskyba) [inline js/hw2.js:113]: Function is fine. But still - take care of edge cases - negative numbers, 0s, string etc.

File: js/hw2.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+}
+
+function guessNumber() {
+    let from = 0;
+    let to = 100;
+    let game_end = false;
+
+    let number;
+
+    while (!game_end) {
```
Comment (roskyba) [inline js/hw2.js:174]: Good algorithm!

File: js/hw2.js (hunk 1 above)
Comment (gbielanova) [reply js/hw2.js:30]: Could be a workaround, but I don't believe that this approach is correct from a math perspective. As this is a learning task I suppose the assumption that input will be a not float number is ok.

File: js/hw2.js (hunk 2 above)
Comment (gbielanova) [reply js/hw2.js:31]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5

File: js/hw2.js (hunk 3 above)
Comment (gbielanova) [reply js/hw2.js:39]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5

File: js/hw2.js (hunk 4 above)
Comment (gbielanova) [reply js/hw2.js:57]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5

File: js/hw2.js (hunk 5 above)
Comment (gbielanova) [reply js/hw2.js:58]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5

File: js/hw2.js (hunk 6 above)
Comment (gbielanova) [reply js/hw2.js:76]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5

File: js/hw2.js (hunk 7 above)
Comment (gbielanova) [reply js/hw2.js:102]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5

File: js/hw2.js (hunk 8 above)
Comment (gbielanova) [reply js/hw2.js:113]: fixed at https://github.com/gbielanova/beetroot_hws_js/pull/5
